src/main/java/uk/gov/hmcts/reform/sscs/services/mapper/CaseDataEventBuilder.java

```diff
@@ -0,0 +1,131 @@
+package uk.gov.hmcts.reform.sscs.services.mapper;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.models.GapsEvent;
+import uk.gov.hmcts.reform.sscs.models.deserialize.gaps2.AppealCase;
+import uk.gov.hmcts.reform.sscs.models.deserialize.gaps2.MajorStatus;
+import uk.gov.hmcts.reform.sscs.models.deserialize.gaps2.MinorStatus;
+import uk.gov.hmcts.reform.sscs.models.serialize.ccd.Event;
+import uk.gov.hmcts.reform.sscs.models.serialize.ccd.Events;
+
+@Service
+class CaseDataEventBuilder {
+
+    List<Events> buildPostponedEvent(AppealCase appealCase) {
+        if (minorStatusIsNotNullAndIsNotEmpty(appealCase.getMinorStatus())) {
+            return appealCase.getMinorStatus().stream()
+                .filter(minorStatus -> areConditionsToCreatePostponedEventMet(minorStatus.getStatusId(), appealCase))
+                .filter(minorStatus -> postponedEventIsNotPresentAlready(minorStatus.getDateSet(),
+                    appealCase.getMajorStatus()))
+                .map(minorStatus -> buildNewPostponedEvent(minorStatus.getDateSet()))
+                .distinct()
+                .collect(Collectors.toList());
+        }
+        return Collections.emptyList();
+    }
+
+    private boolean areConditionsToCreatePostponedEventMet(String statusId, AppealCase appealCase) {
+        //fixme remove minor status id 26 logic
+        if ("26".equals(statusId)) {
+            return true;
+        }
+
+        if (minorStatusIdIs27AndThereIsOnlyOnePostponementRequest(statusId, appealCase)) {
+            return true;
+        }
+
+        if (minorStatusIdIs27AndMoreThanOnePostponementRequest(statusId, appealCase)) {
+            return !appealCase.getPostponementRequests().stream()
+                .filter(postponementRequest -> "Y".equals(postponementRequest.getPostponementGranted()))
+                .filter(postponementRequest -> postponementRequest.getAppealHearingId().equals(
+                    appealCase.getHearing().get(0).getHearingId()))
+                .collect(Collectors.toList())
+                .isEmpty();
+        }
+        return false;
+    }
+
```

Comment:
Why minor status 26 is still being handled?

Reply:
address here -> https://tools.hmcts.net/jira/browse/SSCS-3139

Reply:
done